source/Octopus.Cli/Util/RepositoryExtensions.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Octopus.Cli.Infrastructure;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Extensibility;
+using Octopus.Client.Logging;
+using Octopus.Client.Model;
+using Octopus.Client.Repositories.Async;
+
+namespace Octopus.Cli.Util
+{
+    public static class RepositoryExtensions
+    {
+        private static readonly ILog Logger = LogProvider.GetLogger(typeof(RepositoryExtensions));
+
+        private static async Task<TResource> FindByNameOrIdOrFail<T, TResource>(this T repository,
+            Func<string, Task<TResource>> findByNameFunc, string fixedIdPrefix, string typeDescription, string nameOrId,
+            string inDescription = "", bool skipLog = false)
+            where T : IGet<TResource> where TResource : Resource, INamedResource
+        {
+            TResource resourceById;
+            if (!Regex.IsMatch(nameOrId, $@"^{Regex.Escape(fixedIdPrefix)}-\d+$",
```

Comment:
Move the regex to static fields to gain the compilation benefits

Reply:
I believe static calls should hit the compiled regex cache: https://docs.microsoft.com/en-us/dotnet/standard/base-types/compilation-and-reuse-in-regular-expressions#the-regular-expressions-cache

So I think it's more readable to keep the pattern here at point of use.